src/darwin/Framework/CHIP/MTRDefines_Internal.h

```diff
@@ -152,3 +152,23 @@ typedef struct {} variable_hidden_by_mtr_hide;
 #endif
 
 #define MTR_YES_NO(x) ((x) ? @"YES" : @"NO")
+
+#define _MTR_ABSTRACT_METHOD_THROW(message, ...) \
+    @throw [NSException exceptionWithName:NSInternalInconsistencyException reason:[NSString stringWithFormat:@message, __VA_ARGS__] userInfo:nil]
+
+#define _MTR_ABSTRACT_METHOD_LOG(message, ...) \
+    MTR_LOG_ERROR(message, __VA_ARGS__)
+
+#ifdef DEBUG
+#define _MTR_ABSTRACT_METHOD_IMPL(message, ...)           \
+    do {                                                  \
+        _MTR_ABSTRACT_METHOD_LOG(message, __VA_ARGS__);   \
+        _MTR_ABSTRACT_METHOD_THROW(message, __VA_ARGS__); \
+    } while (0)
+#else // DEBUG
+#define _MTR_ABSTRACT_METHOD_IMPL(message, ...) \
+    _MTR_ABSTRACT_METHOD_LOG(message, __VA_ARGS__)
+#endif // DEBUG
```

Comment:
```suggestion
#ifdef DEBUG
#define _MTR_ABSTRACT_METHOD_IMPL(message, ...)           \
    do {                                                  \
        MTR_LOG_ERROR(message, __VA_ARGS__);   \
        _MTR_ABSTRACT_METHOD_THROW(message, __VA_ARGS__); \
    } while (0)
#else // DEBUG
#define _MTR_ABSTRACT_METHOD_IMPL(message, ...) \
    MTR_LOG_ERROR(message, __VA_ARGS__)
#endif // DEBUG
```


Looks good. I don't think you even need `_MTR_ABSTRACT_METHOD_LOG` anymore since it's just `MTR_LOG_ERROR`